src/content/docs/pl/concepts/islands.mdx

```diff
@@ -0,0 +1,84 @@
+
+<IslandsDiagram>
+  <Fragment slot="headerApp">Nagłówek (interaktywna wyspa)</Fragment>
+  <Fragment slot="sidebarApp">Pasek boczny (statyczny HTML)</Fragment>
+  <Fragment slot="main">Statyczna treść takie jak tekst, obrazy, itp.</Fragment>
+  <Fragment slot="carouselApp">Karuzela obrazów (interaktywna wyspa)</Fragment>
+  <Fragment slot="footer">Stopka (statyczny HTML)</Fragment>
+  <Fragment slot="source">Źródło: [Islands Architecture: Jason Miller](https://jasonformat.com/islands-architecture/)</Fragment>
+</IslandsDiagram>
+
+Wyspa zawsze działa w izolacji od innych wysp na stronie, a na jednej stronie może istnieć wiele wysp. Wyspy nadal mogą dzielić stan i komunikować się ze sobą, mimo że działają w różnych kontekstach komponentów.
+
+Ta elastyczność pozwala Astro na obsługę wielu frameworków UI, takich jak [React](https://react.dev/), [Preact](https://preactjs.com/), [Svelte](https://svelte.dev/), [Vue](https://vuejs.org/) i [SolidJS](https://www.solidjs.com/). Ponieważ są one niezależne, możesz nawet mieszać kilka frameworków na każdej stronie.
+
+:::tip
+Chociaż większość deweloperów będzie trzymać się tylko jednego frameworka UI, Astro obsługuje wiele frameworków w tym samym projekcie. Pozwala to na:
+
+- Wybór frameworka, który jest najlepszy dla każdego komponentu.
+- Naukę nowego frameworka bez konieczności rozpoczynania nowego projektu.
+- Współpracę z innymi, nawet przy pracy z różnymi frameworkami.
+- Stopniowe konwertowanie istniejącej strony na inny framework bez przestojów.
+  :::
+
+## Tworzenie wyspy
+
+Domyślnie Astro automatycznie renderuje każdy komponent UI do samego HTML i CSS, **automatycznie usuwając cały JavaScript po stronie klienta.**
+
+```astro title="src/pages/index.astro"
+<MyReactComponent />
+```
+
+Może to brzmieć restrykcyjnie, ale to zachowanie utrzymuje strony Astro szybkimi domyślnie i chroni deweloperów przed przypadkowym wysyłaniem niepotrzebnego lub niechcianego JavaScript, który mógłby spowolnić ich stronę internetową.
+
+Przekształcenie dowolnego statycznego komponentu UI w interaktywną wyspę wymaga tylko dyrektywy `client:*`. Astro następnie automatycznie buduje i pakuje Twój JavaScript po stronie klienta dla zoptymalizowanej wydajności.
+
+```astro title="src/pages/index.astro" ins="client:load"
+<!-- Ten komponent jest teraz interaktywny na stronie! 
+     Reszta Twojej strony pozostaje statyczna. -->
+<MyReactComponent client:load />
+```
+
+Dzięki wyspom, JavaScript po stronie klienta jest ładowany tylko dla jawnych interaktywnych komponentów, które oznaczysz za pomocą dyrektyw `client:*`.
+
+A ponieważ interakcja jest konfigurowana na poziomie komponentu, możesz obsłużyć różne priorytety ładowania dla każdego komponentu w oparciu o jego wykorzystanie. Na przykład, `client:idle` mówi komponentowi, aby ładował się, gdy przeglądarka staje się bezczynna, a `client:visible` mówi komponentowi, aby ładował się tylko wtedy, gdy wchodzi w obszar widoczności.
```

Comment:
"Na przykład, `client:idle`" na "Na przykład, dyrektywa `client:idle`"
można łatwiej zrozumieć

Reply:
Ale czy to nie jest wiadome na podstawie pierwszego zdania?

Bo jeżeli ja bym dał tak jak ty to napisałeś to bym musiał też to zrobić, dla 'client:visible' i by zrobiło się masło maślane :D